File: modules/day-01-readable/demos/readable-code-bad.R
## Simulated data
df <- matrix(
  c(10, 12, 15, 18, 
    4, 5, 7, 8, 
    20, 19, 18, 16),
  nrow = 3, byrow = TRUE,
  dimnames = list(c("GeneA", "GeneB", "GeneC"),
                  c("S1", "S2", "S3", "S4"))
)

x_var <- c(7.5, 8, 7.7, 6.8)
names(x_var) <- colnames(df)

## The function
take_res <- function(df, x_var) {
  res <- matrix(nrow = 0, ncol = ncol(df))
  colnames(res) <- colnames(df)
  for (i in 1:nrow(df)) {
    model <- lm(df[i,] ~ x_var)
    res <- rbind(res, model$residuals)
  }
  rownames(res) <- rownames(df)
  return(res)
}

## Mock output
mock_output <- take_res(df, x_var)
print(mock_output)

File: modules/day-01-readable/demos/readable-code-bad.py
#%%
import numpy as np
import pandas as pd
import statsmodels.api as sm

#%% Simulated data
df = pd.DataFrame(
    [[10,12,15,18],
     [4,5,7,8],
     [20,19,18,16]],
    index=["GeneA","GeneB","GeneC"],
    columns=["S1","S2","S3","S4"]
)

x_var = pd.Series([7.5,8,7.7,6.8], index=df.columns)

#%% The function
def take_res(df, x_var):
    res = pd.DataFrame(np.empty((0, df.shape[1])))
    res.columns = df.columns
    for i in range(df.shape[0]):
        model = sm.OLS(df.iloc[i,:], sm.add_constant(x_var)).fit()
        res = pd.concat([res, pd.DataFrame([model.resid], columns=df.columns)])
    res.index = df.index
    return res

#%% Test the function
mock_output = take_res(df, x_var)
print(mock_output)
